tools/trap/ramdumpParser.py

```diff
@@ -735,22 +751,48 @@ def find_crash_point(log_file, elf):
     os.system("python3 ../debug/debugsymbolviewer.py " + log_file + " " + str(g_app_idx) + " 1")
     print('-----------------------------------------------------------------------------------------')
 
-# Function to delete the timestamp from the log file if it consists of timestamp at the start of each line
+# Function to format logs and delete the timestamp (format-[xxxxxxxxx]) if it consists of timestamp at the start of each log line
 def format_log_file(log_file):
 
+	# Delete unwanted logs (if any) and timestamp at the start of each line
 	with open(log_file, "r") as f:
 		data = f.readlines()
 	with open(log_file, "w") as f:
+		trunc = True # False if log line is to be retained, True otherwise
 		for line in data:
+			# Truncate logs before up_assert and retain the rest
+			if 'up_assert:' in line:
+				trunc = False
+
 			delete_idx = 0
 			# Timestamp present if line starts with '['
 			if line[0] == '[':
 				for idx in range(0, len(line)):
 					if ']' == line[idx]:
 						delete_idx = idx + 1
 						break
-			line = line[delete_idx:]
-			f.write(line)
+			if line[delete_idx] == ' ':	# Check for trailing white spaces
+				delete_idx = delete_idx + 1
+			if trunc is False:
+				line = line[delete_idx:]
+				f.write(line)
+			else:
```

Comment:
I think this else part can be shifted to above if condition, as we need not check for time stamps for unwanted lines
```
if 'up_assert:' in line:
	trunc = False
else :
	continue
```

Reply:
Thanks. That's a good point. Modified it